meetings/working-groups/extensions/anonymous-extension-declarations.md

```diff
@@ -0,0 +1,368 @@
+
+Two kinds of short-hand syntax are possible:
+
+__Merge static class and extension declarations:__ When a static class contains only a singe extension declaration and nothing else, allow it to be abbreviated to a top-level extension declaration _with_ a name:
+
+``` c#
+public extension(IEnumerable) Enumerable
+{
+    public bool IsEmpty => !GetEnumerator().MoveNext();
+}
+```
+
+This ends up looking more like what we've been calling a "type-based" approach, where the container for extension members is itself named. However, the nesting would still be applied in the generated output.
+
+__Merge extension declaration and extension member:__ When an extension declaration contains only one member, allow the `{ ... }` curlies to be omitted around that member:
+
+``` c#
+public static class Bits
+{
+    extension(ref ulong) public bool this[int index]
+    {
+        get => (this & Mask(index)) != 0;
+        set => this = value ? this | Mask(index) : this & ~Mask(index); // mutates underlying value
+    }
+    static ulong Mask(int index) => 1ul << index;
+}
+```
+``` c#
+public static class Enumerable
+{
+    extension<T>(IEnumerable<T>) public IEnumerable<T> Where(Func<T, bool> predicate) { ... }
+}
+```
+
+This ends up looking more like what we've been calling a "member-based" approach, where each extension member contains its own details about the underlying type. However, in the generated output the type parameter and underlying type would still be applied to a generated nested class, not the member itself.
+
+### Embracing compatibility
+
+Allowing full compatibility for existing extension methods to be ported to new syntax was an explicit non-goal for this proposal, but in the end it does come quite close to being able to embrace such compatibility. an instance extension method in this syntax has everything it needs to generate an equivalent classic extension method - except a parameter name for the `this` parameter. Why is this necessary? Because existing callers of the extension method as a static method may pass the `this` argument by name! 
+
+But herein lies also an opportunity. We could say that extension declarations are _allowed_ to specify a parameter name, and when they do, any extension methods declared within them are generated as static methods on the enclosing static class:
+
+``` c#
+public static class Enumerable
+{
+    extension<TSource>(IEnumerable<TSource> source)
+    {
+        // Parameter name means generate as classic extension method
+        public IEnumerable<TSource> Where(Func<TSource, bool> predicate) 
+        { 
+            foreach (var e in this){ ... }
+        }
+    }
+}
+```
+
+Note that `source` would not be available in member bodies - they still use `this` to refer to the underlying value. The parameter name is purely there to enable - and trigger - generation as a classic extension method into the enclosing static class.
```

Comment:
i'd prefer that if you supply the name, you have to use it.  but that's jus tme :)